### PR TITLE
Create log directory if it doesn't exist

### LINKED ISSUE #157: FileLoggerSink.cpp will fail to create a file if the directory path does not exist.
FileLoggerSink.cpp uses _fwopen/_fwsopen to create the file. This [API](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/fopen-wfopen?view=vs-2019) will fail if the directory path does not exist. 

Consider adding CreateIfNotExist directory check before file open. This has an added benefit of keeping the behavior consistent with our ProductionBreakpoints InstrumentationMethods.

### PR DESCRIPTION
**Pull request by jander-msft**
closes #157

### CODE REVIEW

**Review by WilliamXieMSFT — APPROVED** [review]
:shipit:

File: src/InstrumentationEngine.Lib/FileLoggerSink.cpp [hunk 1]
```diff
@@ -75,45 +75,55 @@ HRESULT CFileLoggerSink::Reset(_In_ LoggingFlags defaultFlags, _Out_ LoggingFlag
             wszExtension,
             _MAX_EXT);
 
-        tstring tsFile(wszFile);
-        tstring tsExtension(wszExtension);
+        // Try to create log directory if it doesn't exist
+        BOOL fDirectoryExists = PathFileExists(wszDirectory);
+        if (!fDirectoryExists)
+        {
+            fDirectoryExists = CreateDirectory(wszDirectory, nullptr);
```
Comment (wiktork) [inline src/InstrumentationEngine.Lib/FileLoggerSink.cpp:82]: It looks like CreateDirectory can return 0, with an ERROR_ALREADY_EXISTS error code. CreateDirectory also doesn't handle intermediate paths. You probably want to check both of those error codes.

**Review by wiktork — CHANGES_REQUESTED** [review]
Need additional handling for CreateDirectory error codes.

**Review by delmyers — APPROVED** [review]
:shipit:

**Comment by jander-msft** [comment]
As @wiktork pointed out, there's more work here to get the issue completely fixed. I'll abandon at this time and create a new PR when ready.

**Pull request closed without merging**